bin.js

```diff
@@ -121,27 +122,48 @@ yargs.command('run <system> [location] [test]', 'Run your package\'s tests in re
     // Add the raw package.json template
     const rnPkg = require(`${root}/template-package.json`)
     await writeFile(`${root}/package.json`, JSON.stringify(rnPkg, null, 2) + '\n')
-    try {
-      // Install any dependencies of the wrapper app
-      await logExec('npm', ['i'], { cwd: root, quiet: !verbose })
-    } finally {
-      // Add the package name to dependencies so it will compile
-      rnPkg.dependencies[packageJSON.name] = ''
-      await writeFile(`${root}/package.json`, JSON.stringify(rnPkg, null, 2) + '\n')
-    }
 
-    // Make sure the folder exists for copying over the package
-    console.log(`## react-native:preparing-dep [target=${target}]`)
-    await mkdirp(path.dirname(target))
+    // Install any dependencies of the wrapper app
+    await logExec('npm', ['i'], { cwd: root, quiet: !verbose })
+
+    console.log('## react-native:npm pre-pack')
+
+    // Merge devDependencies with dependencies
+    const { dependencies = {}, devDependencies = {} } = packageJSON
+    const combinedDeps = { ...dependencies, ...devDependencies }
+    const modifiedPackageJSON = { ...packageJSON, dependencies: combinedDeps }
+    await writeFile(packageJSONLocation, JSON.stringify(modifiedPackageJSON))
 
-    // Copy over the package contents
-    console.log(`## react-native:copy-build [src=${packageLocation} target=${target}]`)
+    console.log('## react-native:npm pack')
 
-    await cpr(packageLocation, target, {
-      // Avoid recursive copying
-      filter: (item) => !item.startsWith(`${__dirname}/`)
+    // Pack up package
+    await logExec('npm', ['pack'], {
+      cwd: packageLocation,
+      quiet: !verbose
     })
 
+    // Get a reference to the tar file
+    const packageFiles = await readdir(packageLocation)
+    const matchTargz = new RegExp(`^${packageJSON.name}-.*\\.tgz$`)
+    const tarName = packageFiles.find((name) => name.match(matchTargz))
+
+    if (!tarName) throw new Error('Tar file for package could not be generated')
+
+    const tarPath = path.join(packageLocation, tarName)
+
+    // Restore old package.json
+    await writeFile(packageJSONLocation, packageRaw)
```

Comment:
This should be in a `try { } finally {}` block to make sure that the `package.json` is definitely restored!

Reply:
Good catch!